src/engine/SpatialJoin.cpp

```diff
@@ -0,0 +1,295 @@
+
+
+string SpatialJoin::getCacheKeyImpl() const {
+  if (childLeft && childRight) {
+    std::ostringstream os;
+    os << "SpatialJoin\nChild1:\n" << childLeft->getCacheKey() << "\n";
+    os << "Child2:\n" << childRight->getCacheKey() << "\n";
+    return std::move(os).str();
+  } else {
+    return "incomplete SpatialJoin class";
+  }
+}
+
+string SpatialJoin::getDescriptor() const {
+  return "Descriptor of SpatialJoin";
+}
+
+size_t SpatialJoin::getResultWidth() const {
+  if (childLeft && childRight) {
+    // don't subtract anything because of a common join column. In the case
+    // of the spatial join, the join column is different for both sides (e.g.
+    // objects with a distance of at most 500m to each other, then each object
+    // might contain different positions, which should be kept)
+    if (addDistToResult) {
+      return childLeft->getResultWidth() + childRight->getResultWidth() + 1;
+    } else {
+      return childLeft->getResultWidth() + childRight->getResultWidth();
+    }
+  } else {
+    return 1;  // dummy return, as the class does not have its children yet
+  }
+}
+
+size_t SpatialJoin::getCostEstimate() {
+  return 50000;  // dummy return for now
+}
+
+uint64_t SpatialJoin::getSizeEstimateBeforeLimit() {
+  return 1;  // dummy return for now
+}
+
+float SpatialJoin::getMultiplicity(size_t col) {
+  return 1;  // dummy return for now
+}
+
+bool SpatialJoin::knownEmptyResult() {
+  return false;  // dummy return for now
+}
+
+vector<ColumnIndex> SpatialJoin::resultSortedOn() const {
+  return {};  // dummy return for now
+}
+
+ResultTable SpatialJoin::computeResult() {
+  auto betweenQuotes = [] (std::string extractFrom) {
+    /* returns everything between the first two quotes. If the string does not
+    * contain two quotes, the string is returned as a whole
+    */
+    size_t pos1 = extractFrom.find("\"", 0);
+    size_t pos2 = extractFrom.find("\"", pos1 + 1);
+    if (pos1 != std::string::npos && pos2 != std::string::npos) {
+      return extractFrom.substr(pos1 + 1, pos2-pos1-1);
+    } else {
+      return extractFrom;
+    }
+  };
+  
+  auto computeDist = [&](const IdTable* res_left, const IdTable* res_right,
+                        size_t rowLeft, size_t rowRight,
+                        ColumnIndex leftPointCol, ColumnIndex rightPointCol) {
+    std::string point1 = ExportQueryExecutionTrees::idToStringAndType(
+            getExecutionContext()->getIndex(),
+            res_left->at(rowLeft, leftPointCol), {}).value().first;
+    std::string point2 = ExportQueryExecutionTrees::idToStringAndType(
+            getExecutionContext()->getIndex(),
+            res_right->at(rowRight, rightPointCol), {}).value().first;
+    point1 = betweenQuotes(point1);
+    point2 = betweenQuotes(point2);
+    double dist = ad_utility::detail::wktDistImpl(point1, point2);
+    return dist * 1000;  // convert to meters
+  };
+
+  // a maximum distance of 0 encodes infinity -> return cross product
+  if (maxDist == 0) {
```

Comment:
As said: We don't need the infinity case. Just assert during the parsing, that the distance is strictly positive and a valid integer. This makes this code simpler, because you don't need this code.